packages/app/src/app/pages/SignIn/ChooseProvider.tsx

```diff
@@ -8,23 +17,198 @@ import {
 } from '@codesandbox/components/lib/components/Icon/icons';
 import { useActions, useAppState } from 'app/overmind';
 import history from 'app/utils/history';
-
 import { Button } from './components/Button';
 
+const StyledHeading = styled(Text)`
+  margin: 0;
+  text-align: center;
+  font-family: Everett, sans-serif;
+  font-size: 48px;
+  font-weight: 500;
+  line-height: 1.17;
+  letters-pacing: -0.018em;
+`;
+
+const StyledGhostButton = styled(MainButton)`
+  padding: 4px;
+  width: fit-content;
+
+  :hover:not([disabled]),
+  :focus:not([disabled]) {
+    background: transparent;
+  }
+`;
+
+type ValidationState =
+  | { state: 'IDLE' }
+  | { state: 'VALIDATING' }
+  | { state: 'VALID' }
+  | { state: 'INVALID'; error: string };
+
+type SSOSignInProps = {
+  changeSignInMode: () => void;
+};
+const SSOSignIn: React.FC<SSOSignInProps> = ({ changeSignInMode }) => {
+  const inputRef = React.useRef<HTMLInputElement>(null);
+  const [validationState] = React.useState<ValidationState>({
+    state: 'IDLE',
+  });
+
+  React.useEffect(() => {
+    const inputElement = inputRef.current;
+    if (inputElement) {
+      inputElement.focus();
+    }
+  }, []);
+
+  const handleFormSubmit = React.useCallback(
+    async (e: FormEvent<HTMLFormElement>) => {
+      e.preventDefault();
+    },
+    [validationState]
+  );
+
+  return (
+    <Stack
+      as={motion.div}
+      align="center"
+      initial={{ opacity: 0 }}
+      animate={{ opacity: 1 }}
+      exit={{ opacity: 0 }}
+      direction="vertical"
+      gap={8}
+    >
+      <StyledHeading as="h1" id="heading">
+        Sign in with SSO login
+      </StyledHeading>
+      <Stack
+        as="form"
+        css={{ width: '100%', maxWidth: '420px' }}
+        direction="vertical"
+        onSubmit={handleFormSubmit}
+        gap={2}
+      >
+        <Stack direction="vertical" gap={1}>
+          <Element css={{ position: 'relative', height: '48px' }}>
+            <Input
+              aria-describedby="heading error"
```

Comment:
```suggestion
              aria-labelledby="heading"
              aria-describedby="error"
```

There are some more things we can add or improve regarding a11y, but I would love to bake that into the components one day.